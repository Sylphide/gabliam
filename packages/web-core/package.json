{
  "name": "@gabliam/web-core",
  "version": "7.9.4",
  "description": "Gabliam plugin for add web-core",
  "keywords": [
    "gabliam",
    "gabliam-plugin",
    "plugin",
    "web-core",
    "rest"
  ],
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "module": "lib/index.js",
  "scripts": {
    "build": "rimraf lib && tsc -p tsconfig.build.json"
  },
  "files": [
    "lib",
    "src"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gabliam/gabliam.git"
  },
  "author": "David Touzet <dtouzet@gmail.com>",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/gabliam/gabliam/issues"
  },
  "homepage": "https://github.com/gabliam/gabliam/tree/master/packages/web-core#readme",
  "peerDependencies": {
    "@gabliam/core": "^7.2.0"
  },
  "devDependencies": {
    "@gabliam/core": "7.9.4",
    "@types/escape-html": "^0.0.20"
  },
  "dependencies": {
    "escape-html": "^1.0.3",
    "http-status-codes": "^1.2.0",
    "tslib": "^1.8.0"
  },
  "engines": {
    "node": ">=8.9.0"
  }
}
