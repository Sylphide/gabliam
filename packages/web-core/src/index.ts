export * from './decorators';
export * from './constants';
export * from './gab-context';
export * from './gab-request';
export * from './gab-response';
export * from './plugin-config';
export * from './response-entity';
export * from './utils';
export * from './validate-request';
export * from './web-configuration';
export * from './web-plugin-base';
