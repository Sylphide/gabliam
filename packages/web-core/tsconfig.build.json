{
  "extends": "../../tsconfig",
  "compilerOptions": {
    "sourceMap": false,
    "outDir": "lib"
  },
  "include":[
    "src/**/*.ts"
  ]
}
