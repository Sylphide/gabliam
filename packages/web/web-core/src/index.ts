export * from './constants';
export * from './cookies';
export * from './decorators';
export * from './errors';
export * from './execution-context';
export * from './gab-context';
export * from './gab-request';
export * from './gab-response';
export * from './interface';
export * from './plugin-config';
export * from './response-entity';
export * from './utils';
export * from './validate';
export * from './web-configuration';
export * from './web-plugin-base';
