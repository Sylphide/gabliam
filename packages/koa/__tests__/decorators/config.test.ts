import { WebConfigMetadata, METADATA_KEY, WebConfig } from '@gabliam/web-core';

describe('WebConfig decorator', () => {
  test('should add WebConfig metadata to a class when decorated with @WebConfig', () => {
    class TestConfig {
      @WebConfig()
      testMethod() {}

      @WebConfig()
      test2Method() {}
    }

    const webConfigMetadata: WebConfigMetadata[] = Reflect.getMetadata(
      METADATA_KEY.webConfig,
      TestConfig
    );

    expect(webConfigMetadata).toMatchSnapshot();
  });

  test('should add WebConfig metadata to a class when decorated with @WebConfig(order)', () => {
    class TestConfig {
      @WebConfig(50)
      testMethod() {}

      @WebConfig(100)
      test2Method() {}
    }

    const webConfigMetadata: WebConfigMetadata[] = Reflect.getMetadata(
      METADATA_KEY.webConfig,
      TestConfig
    );

    expect(webConfigMetadata).toMatchSnapshot();
  });

  test('should fail when decorated multiple times the same method with @WebConfig', () => {
    expect(function() {
      class TestConfig {
        @WebConfig()
        @WebConfig(50)
        testMethod() {}

        @WebConfig(100)
        test2Method() {}
      }
      // tslint:disable-next-line:no-unused-expression
      new TestConfig();
    }).toThrowError();
  });
});
