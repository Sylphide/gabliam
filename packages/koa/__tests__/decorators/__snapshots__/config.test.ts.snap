// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebConfig decorator should add WebConfig metadata to a class when decorated with @WebConfig 1`] = `
Array [
  Object {
    "id": "TestConfig",
    "key": "testMethod",
    "order": 1,
  },
  Object {
    "id": "TestConfig",
    "key": "test2Method",
    "order": 1,
  },
]
`;

exports[`WebConfig decorator should add WebConfig metadata to a class when decorated with @WebConfig(order) 1`] = `
Array [
  Object {
    "id": "TestConfig",
    "key": "testMethod",
    "order": 50,
  },
  Object {
    "id": "TestConfig",
    "key": "test2Method",
    "order": 100,
  },
]
`;
